### PR TITLE
Inserted Guidelines to maintainers

### PR DESCRIPTION
**Pull request by sergiomario**
I could open an issue to discuss this, but I have adopted a philosophy of starting a discussion with a proposal.

I created this chapter to propose an organization for the project maintainers. The idea is that it is not legislation with multiple responsibilities, but a guide that facilitates communication and reduces rework.

I do this when we invite @giuliocc to compose our team of maintainers. :1st_place_medal:  Thus, we are growing and creating a working model to prevent distress to people as generous as Querido Diario maintainers.

"There is a criteria I call change latency, which is the round-trip time from identifying a problem to testing a solution. The faster the better. If maintainers cannot respond to pull requests as rapidly as people expect, they're not doing their job (or they need more hands)." (Peter Hintjens)
https://hintjens.gitbooks.io/social-architecture/content/chapter4.html

I would like opinions: @jvanz @rennerocha

### CODE REVIEW

**Comment by jvanz** [comment]
I would like to raise two topics for discussion:

1. Squash commits: My personal preference is to have a clean git history. Thus, when we have a git branch with a lot of commits fixing review issues, we should remember to squash the commit. Leaving the meaningful ones only.
 
2. Should we require more then one reviewer? I do not know if this is realistic for the current state of the project. But I think we should give opportunity to the community to review other members PRs. This can be a soft rule. It does not need to be a hard rule to get the PR merged.

**Comment by rennerocha** [comment]
I don't think that Squash all PRs is the best policy. It is a good idea when we are adding a brand new spider but sometimes, if the contributor has a good discipline, the commits can be separately well defined. Probably it may be a soft rule, but when we are merging we can choose just to rebase if we consider that the commit history is good as it is.

As an example I can mention this [PR](https://github.com/okfn-brasil/querido-diario/pull/252/commits). One commit is adding the formatting configurations and a second commit is applying the formatting into the codebase. I'd prefer to have a separated commit for this because if I want to understand when automatic formatting was added, it will be a mess to look into all changes in the codebase until I find what I am looking for.

**Comment by rennerocha** [comment]
>     2. Should we require more then one reviewer? I do not know if this is realistic for the current state of the project. But I think we should give opportunity to the community to review other members PRs. This can be a soft rule. It does not need to be a hard rule to get the PR merged.

I don't think we need this for now.

**Comment by rennerocha** [comment]
>     3. The maintainers should run all the spiders before merging it.

Agreed!!! I reviewed several spiders that the contributor certainly didn't run the spider entirely once.

**Comment by rennerocha** [comment]
@sergiomario 
I didn't understand the labeling rules very well. In my opinion it will be good to have a well defined set of labels and documented rules when to use them.

**Comment by jvanz** [comment]
> > ```
> > 1. Squash commits: My personal preference is to have a clean git history. Thus, when we have a git branch with a lot of commits fixing review issues, we should remember to squash the commit. Leaving the meaningful ones only.
> > ```
> 
> I don't think that Squash all PRs is the best policy. It is a good idea when we are adding a brand new spider but sometimes, if the contributor has a good discipline, the commits can be separately well defined. Probably it may be a soft rule, but when we are merging we can choose just to rebase if we consider that the commit history is good as it is.
> 
> As an example I can mention this [PR](https://github.com/okfn-brasil/querido-diario/pull/252/commits). One commit is adding the formatting configurations and a second commit is applying the formatting into the codebase. I'd prefer to have a separated commit for this because if I want to understand when automatic formatting was added, it will be a mess to look into all changes in the codebase until I find what I am looking for.

I agree with you. Sorry, I would not  sound like "squash everything". As I said `Leaving the meaningful ones only.`  I just wanna avoid is like "Fix typo1", "Fix typo2", "Fix typo3" or a generic "Review fixes". Any way, I think we are in the same page.

**Comment by rougeth** [comment]
Hey folks, I was reading your comments and found this topic super relevante/interesting. I have one question about @jvanz's comment:

> The maintainers should run all the spiders before merging it.

It should be possible to run the spider in a Github Action (or other CIs), right? Github API allows to see which files were changed in a Pull Request, so there's a way to know which spiders were changed. What do you think about it?

**Comment by rennerocha** [comment]
> It should be possible to run at least part of the spider that validates the code change in a Github Action, right?

Is it? I have no idea :smile: Never worked with Github Actions so I don't know if this is possible or not. But it would be a great improvement if we can setup a basic environment with Scrapy that runs the spider that changed with files download disabled and check some basics stats (if items were scraped, if it has no errors, etc). Maybe Travis can do something like that?

Anyway, asking for the maintainers to run the spiders locally as well is still a good advice ...

**Comment by jvanz** [comment]
Yeah, I think is possible to run the spider in Github Actions. That's my idea when I created the issue #148. But I think that I was not clear enough. We can try again. Open a issue for that with more info about what we want and how to consider a spider okay or not. ;-)

**Comment by sergiomario** [comment]
Hi firends. :)

I updated the guidelines.
- Removed labels control - I think that as funcionalidades de fluxo de revisão de código que existem no Github resolvem os problemas que eu havia pensado em resolver com as labels de controle.
- "The maintainers should run all the spiders before merging it." @jvanz :+1: 
- Squash and merge with the caveats described by @rennerocha

**Review by jvanz — APPROVED** [review]

**Review by rennerocha — APPROVED** [review]

**Pull request merged**
